### PR TITLE
Render-messages: usernames

### PR DESCRIPTION
**Pull request by camachom**
This PR is for issue is for the first task of issue #1 
This was my approach:
1. join all messages for a given date
2. parse through all the messages and get the user-ids
3. query the db once to get the needed user-names 
3. replace the user-ids with user-names

I'm using two variations of somewhat similar regex because the different groupings make it easier to work in some situations : `(?<=<@).*?(?=>)` `<@(.*?)>`, but maybe I should stick to one.

Looking forward to your feedback.

### CODE REVIEW

File: src/clj/clojurians_log/components/parser.clj [hunk 1]
```diff
@@ -0,0 +1,23 @@
+(ns clojurians-log.components.parser
```
Comment (plexus) [inline src/clj/clojurians_log/components/parser.clj]: `clojurians-log.components.*` namespaces are for implementing components in the [Component](https://github.com/stuartsierra/component) sense. 

Naming is always hard... Maybe `clojurians-log.slack-messages` or `clojurians-log.slack.messages`?

File: src/clj/clojurians_log/components/parser.clj [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  (:require [clojure.string :as str]))
+
+(defn- extract-texts
+  [messages]
+  (let [texts (map :message/text messages)]
+    (clojure.string/join " " texts)))
```
Comment (plexus) [inline src/clj/clojurians_log/components/parser.clj]: you're already aliasing `clojure.string` to `str`, so might as well make this `(str/join ,,,)`

File: src/clj/clojurians_log/components/parser.clj [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+(defn- parse-users
+  [texts]
+  (re-seq #"(?<=<@).*?(?=>)" texts))
+
+(defn extract-user-ids
+  "returns all the user/slack-ids from a string"
+  [messages]
+  (parse-users (extract-texts messages)))
```
Comment (plexus) [inline src/clj/clojurians_log/components/parser.clj]: You're joining all these messages into a string so you can let the regex do its thing, I would avoid building up that big string.

```
(mapcat (comp parse-users :message/text) messages)
```

You also don't care about duplicates, which means maybe this should be a set, which means you'll be using `into`, which means you might as well use a transducer.

```
(into #{} (comp
           (map :message/text)
           (map parse-users)
           cat)
      messages)
```

File: src/clj/clojurians_log/db/queries.clj [hunk 4]
```diff
@@ -55,3 +55,13 @@
          [?chan :channel/name ?chan-name]]
        db
        name))
+
+(defn user-names
+  [db names]
+  (d/q '[:find ?ids ?username
+         :in $ [?ids ...]
```
Comment (plexus) [inline src/clj/clojurians_log/db/queries.clj]: Since you're destructuring this collection the variable will refer to a single `?id`, not multiple `?ids`

**Comment by plexus** [comment]
By the way I already pushed your changes to http://clojurians-log-staging.clojureverse.org, e.g. http://clojurians-log-staging.clojureverse.org/clojure/2015-06-16.html . Looks good :)

**Comment by jxa** [comment]
I was looking at how to parse the slack messages with instaparse the other day... I'll get it into PR shape and maybe we can merge our efforts!

**Comment by jxa** [comment]
I have opened a PR here https://github.com/clojureverse/clojurians-log-app/pull/11

**Comment by camachom** [comment]
thanks for the awesome feedback!

File: src/clj/clojurians_log/slack_messages.clj [hunk 5]
```diff
@@ -0,0 +1,27 @@
+(ns clojurians-log.slack-messages
+  (:require [clojure.string :as str]))
+
+(defn- extract-texts
```
Comment (plexus) [inline src/clj/clojurians_log/slack_messages.clj]: this is no longer needed now.

**Pull request merged**